### DISCUSSION TITLE
Are dynamic favicons possible with Next.js?

### DISCUSSION THREAD

**Discussion by jphyqr** [vercel/vercel#10118] [Ideas] [answered]
### Idea

I always have too many tabs open and end up just seeing favicons.

Imagine for each localhost a little icon that was say 00, 01, 02.. etc

Imagine instead of 10 jira logos it was icons that were just the ticket numbers

![Screen Shot 2023-06-18 at 1 14 55 PM](https://github.com/vercel/vercel/assets/34660850/13ebaa16-bd3a-4b03-991e-9553706bf067)

The dynamic twitter/og image apis have me wondering if you could make your site have a dynamic favicons

seems like it would be great for lots of apps

**Comment by leerob** [comment, marked as answer]
Yes, this is possible with Next.js.

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#generate-icons-using-code-js-ts-tsx
